common/lib/xmodule/xmodule/modulestore/courseware_index.py

```diff
@@ -127,9 +130,50 @@ def remove_index_item_location(item_location):
         if raise_on_error and error_list:
             raise SearchIndexingError(_('Error(s) present during indexing'), error_list)
 
+        return indexed_count
+
+    @classmethod
+    def do_publish_index(cls, modulestore, location, delete=False, raise_on_error=False):
+        """
+        Add to courseware search index published section and children
+        """
+        indexed_count = cls.add_to_search_index(modulestore, location, delete, raise_on_error)
+        cls._track_index_request('edx.course.index.published', indexed_count, str(location))
+        return indexed_count
+
     @classmethod
     def do_course_reindex(cls, modulestore, course_key):
         """
         (Re)index all content within the given course
         """
-        return cls.add_to_search_index(modulestore, course_key, delete=False, raise_on_error=True)
+        indexed_count = cls.add_to_search_index(modulestore, course_key, delete=False, raise_on_error=True)
+        cls._track_index_request('edx.course.index.reindexed', indexed_count)
+        return indexed_count
+
+    @staticmethod
+    def _track_index_request(event_name, indexed_count, location=None):
+        """Track content index requests.
+
+        Arguments:
+            location (str): The ID of content to be indexed.
+            event_name (str):  Name of the event to be logged.
+        Returns:
+            None
+
+        """
+
+        from eventtracking import tracker as track
+        tracker = track.get_tracker()
+        tracker.resolve_context()  # pylint: disable=no-member
```

Comment:
You should be able to replace all three of these lines with simply

```python
from eventtracking import tracker
```

Later you can then just call `tracker.emit()` directly.